x-pack/agent/pkg/agent/application/filters/constraints_filter.go

```diff
@@ -0,0 +1,204 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package filters
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/Masterminds/semver"
+
+	"github.com/elastic/beats/v7/x-pack/agent/pkg/agent/application/info"
+	"github.com/elastic/beats/v7/x-pack/agent/pkg/agent/errors"
+	"github.com/elastic/beats/v7/x-pack/agent/pkg/agent/transpiler"
+	"github.com/elastic/beats/v7/x-pack/agent/pkg/boolexp"
+	"github.com/elastic/beats/v7/x-pack/agent/pkg/release"
+	"github.com/elastic/go-sysinfo"
+)
+
+const (
+	datasourcesKey          = "datasources"
+	constraintsKey          = "constraints"
+	validateVersionFuncName = "validate_version"
+)
+
+const (
+	agentIDKey      = "agent.id"
+	agentVersionKey = "agent.version"
+	hostArchKey     = "host.architecture"
+	osFamilyKey     = "os.family"
+	osKernelKey     = "os.kernel"
+	osPlatformKey   = "os.platform"
+	osVersionKey    = "os.version"
+)
```

Comment:
Can you add a documentation with the variables that are accessible in the configuration?
This will be useful when writing the documentation and also when we implement them in UI.